@woocommerce/components dependency breaks tests for external plugins

## **Describe the bug**

I'm writing a plugin the depends on `@woocommerce/components`, and testing it using `wp-scripts`.
When any of my test files imports (directly or indirectly) `@woocommerce/components` tests fail with:
```
Jest encountered an unexpected token
...
    Details:

    /home/tomalec/repos/wc-com-test/node_modules/@woocommerce/components/build/index.js:4
    import 'react-dates/initialize'; // The above: Turn on react-dates classes/styles, see https://github.com/airbnb/react-dates#initialize
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module
```

I tried tweaking with `transformIgnorePatterns`, but that resulted in even stranger:
```
    Cannot find module '@wordpress/url' from 'index.js'

    Require stack:
      node_modules/@woocommerce/components/node_modules/@woocommerce/navigation/build/index.js
      node_modules/@woocommerce/components/build/advanced-filters/index.js
      node_modules/@woocommerce/components/build/index.js
      js/src/test/fails.js

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:299:11)
      at Object.<anonymous> (../../../../Users/psealock/vagrant-local/www/tangaroa/public_html/wp-content/plugins/wc-admin/packages/navigation/src/index.js:4:1)

Test Suites: 1 failed, 1 passed, 2 total
```
or
```
    TypeError: Cannot read property 'add' of undefined

      at setupWPTimezone (node_modules/@woocommerce/components/node_modules/@wordpress/date/build/@wordpress/date/src/index.js:141:15)
      at Object.<anonymous> (node_modules/@woocommerce/components/node_modules/@wordpress/date/build/@wordpress/date/src/index.js:527:1)
      at Object.<anonymous> (../../../../Users/psealock/vagrant-local/www/tangaroa/public_html/wp-content/plugins/wc-admin/packages/components/src/date/index.js:5:1)
      at Object.<anonymous> (../../../../Users/psealock/vagrant-local/www/tangaroa/public_html/wp-content/plugins/wc-admin/packages/components/src/index.js:14:1)
      at Object.<anonymous> (js/src/setup-mc/top-bar/index.js:4:1)
      at Object.<anonymous> (js/src/setup-mc/index.js:4:1)
      at Object.<anonymous> (js/src/setup-mc/test/index.js:9:1)
```

## **To Reproduce**
Steps to reproduce the behavior:

1. Set up a repo that uses `@wordpress/scripts` and `@woocommerce/components`
2. Write a test that uses `@woocommerce/components`
	```js
	import foo from '@woocommerce/components';

	describe( 'Dependencies', () => {
		test( 'should not break tests', () => {
			expect( true ).toEqual( true );
		   } );
	} );
	```
3. Run the tests: `wp-scripts test-unit-js`

## **Expected behavior**
The test should run and pass

## **Screenshots**
![image](https://user-images.githubusercontent.com/17435/109677992-7c813080-7b7a-11eb-89b5-8ed25b01c8b6.png)


<details>
<summary>Full log</summary>
<pre><code>
> wp-scripts test-unit-js

 PASS  js/src/test/works.js
  Dependencies
    ✓ should not break tests (3ms)

 FAIL  js/src/test/fails.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    /home/tomalec/repos/wc-com-test/node_modules/@woocommerce/components/build/index.js:4
    import 'react-dates/initialize'; // The above: Turn on react-dates classes/styles, see https://github.com/airbnb/react-dates#initialize
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      1 | // Any of this breaks
    > 2 | import foo from '@woocommerce/components';
        | ^
      3 |
      4 | describe( 'Dependencies', () => {
      5 | 	test( 'should not break tests', () => {

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1258:14)
      at Object.<anonymous> (js/src/test/fails.js:2:1)

Test Suites: 1 failed, 1 passed, 2 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.355s
Ran all test suites.

</code></pre>


</details>

## **Additional context**

The `transformIgnorePatterns` tweak I tried:
```js
const defaultConfig = require( '@wordpress/scripts/config/jest-unit.config' );

module.exports = {
	...defaultConfig,
	// That does not help either
	transformIgnorePatterns: [ ...(defaultConfig.transformIgnorePatterns || []), 'node_modules/(?!@woocommerce)' ],
};

```

--- a/packages/components/CHANGELOG.md
+++ b/packages/components/CHANGELOG.md
@@ -3,6 +3,7 @@
 -   Replace deprecated wp.compose.withState with wp.element.useState. #8338
 -   Add missing dependencies. #8349
 -   Update all js packages with minor/patch version changes. #8392
+-   Add moment-timezone to package.json. #6483
 
 # 9.0.0
 

--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -61,6 +61,7 @@
 		"gridicons": "^3.4.0",
 		"memoize-one": "^5.2.1",
 		"moment": "^2.29.1",
+		"moment-timezone": "^0.5.34",
 		"prop-types": "^15.8.1",
 		"react-dates": "^17.2.0",
 		"react-router-dom": "^5.3.0",

